src/stan/callbacks/json_writer.hpp

```diff
@@ -287,12 +286,12 @@ class json_writer {
   void end_record() {
     if (output_ == nullptr)
       return;
-    *output_ << "}";
     record_depth_--;
     if (record_depth_ > 0) {
       record_needs_comma_ = true;
     }
     record_element_needs_comma_ = false;
+    *output_ << "}";
```

Comment:
Is there a reason we don't just add a newline here like 

```suggestion
    *output_ << "}\n";
```

instead of having a new function? Or do we need `std::endl`?

Reply:
that's another option - no objections to that either.

Reply:
the output files end up looking funny because the logic to insert the comma between records is invoked elsewhere,
resulting in "}\n," instead of "},\n".    I wasted an awful lot of time trying to make this pretty.  not worth it.

Reply:
imo I'd just like to not complicate the interface with a new function. if `},\n` is valid json then I think that's fine

Reply:
yes, "}\n," is valid JSON.

@WardBrian - what do you think?

Reply:
My preference is also for simpler, so if it can be done without another function great. If the `}\n,` is aesthetically un-pleasing, it's also reasonable to store an extra bool called `should_newline` and then make the actual printing of `\n` part of the comma logic/end of file logic?